File: server/write/spring/src/main/kotlin/com/board/server/write/spring/models/BoardReadData.kt
package com.board.server.write.spring.models

import com.board.server.write.spring.entities.BoardEntity
import java.time.LocalDateTime
import java.util.*

data class BoardReadData(private val board: BoardEntity) {
    val uuid: UUID = board.uuid
    val uniqueId: String = board.uniqueId
    val name: String = board.name
    val description: String = board.description
    val createdAt: LocalDateTime = board.createdAt
    val updatedAt: LocalDateTime = board.updatedAt
}

File: server/read/fastify/src/models/BoardReadData.ts
import { Board } from '../entities/Board';

export class BoardReadData {
  uuid: string;
  uniqueId: string;
  name: string;
  description: string;
  createdAt: Date;
  updatedAt: Date;

  constructor(board: Board) {
    this.uuid = board.uuid;
    this.uniqueId = board.uniqueId;
    this.name = board.name;
    this.description = board.description;
    this.createdAt = board.createdAt;
    this.updatedAt = board.updatedAt;
  }
}
